src/network/networkd-util.c

```diff
@@ -10,46 +10,83 @@
 #include "string-util.h"
 #include "web-util.h"
 
-static const char* const address_family_table[_ADDRESS_FAMILY_MAX] = {
+static const char * const network_config_source_table[_NETWORK_CONFIG_SOURCE_MAX] = {
+        [NETWORK_CONFIG_SOURCE_FOREIGN] = "foreign",
+        [NETWORK_CONFIG_SOURCE_STATIC]  = "static",
+        [NETWORK_CONFIG_SOURCE_IPV4LL]  = "IPv4LL",
+        [NETWORK_CONFIG_SOURCE_DHCP4]   = "DHCP4",
+        [NETWORK_CONFIG_SOURCE_DHCP6]   = "DHCP6",
+        [NETWORK_CONFIG_SOURCE_DHCP6PD] = "DHCP6PD",
+        [NETWORK_CONFIG_SOURCE_NDISC]   = "NDisc",
```

Comment:
hmm, is there a reason why this use uppercase chars? most of our enum string tables use lower case strings.

Reply:
Well, this is used in log messages, and not used in any settings. So, I think upper case is better.

(I noticed that maybe we should use `DHCPv4` instead of `DHCP4`. I will update so.)

Reply:
maybe add a comment about the use of this, to clarify this for the next reader?

Reply:
Done.